source/DataBase/QueryOperationBuilder.cpp

```diff
@@ -0,0 +1,129 @@
+
+#include "QueryOperationBuilder.hpp"
+
+#include <format>
+
+#include "common/DataBaseCommon.hpp"
+namespace dataBase {
+namespace {
+std::unordered_map<common::DataBaseRequest::TableType, std::string> tableNames =
+    {{common::DataBaseRequest::TableType::Users, common::users},
+     {common::DataBaseRequest::TableType::Passwords, common::passwords},
+     {common::DataBaseRequest::TableType::PasswordHistory,
+      common::passwordHistory},
+     {common::DataBaseRequest::TableType::Categories, common::categories}};
+
+std::string getTableName(common::DataBaseRequest::TableType table) {
+  if (auto it = tableNames.find(table); it != tableNames.end())
+    return it->second;
+
+  return {};
+}
+
+bool isDataEmpty(const common::RequestData &data) {
+  return data.empty() ? true : false;
+}
+
+bool isPrimaryKeyEmpty(const common::PrimaryKey &primaryKey) {
+  return (primaryKey.first.empty() || primaryKey.second.empty()) ? true : false;
+}
+}  // namespace
+QueryOperationBuilder::QueryOperationBuilder() {
+  queryBuilders_ = {{common::DataBaseRequest::OperationType::Insert,
+                     std::bind(&QueryOperationBuilder::buildInsertQuery, this,
+                               std::placeholders::_1, std::placeholders::_2)},
+                    {common::DataBaseRequest::OperationType::Delete,
+                     std::bind(&QueryOperationBuilder::buildDeleteQuery, this,
+                               std::placeholders::_1, std::placeholders::_2)},
+                    {common::DataBaseRequest::OperationType::Update,
+                     std::bind(&QueryOperationBuilder::buildUpdateQuery, this,
+                               std::placeholders::_1, std::placeholders::_2)},
+                    {common::DataBaseRequest::OperationType::Select,
+                     std::bind(&QueryOperationBuilder::buildSelectQuery, this,
+                               std::placeholders::_1, std::placeholders::_2)}};
+}
+std::string QueryOperationBuilder::buildQuery(
+    const common::DataBaseRequest &requestData) {
+  if (requestData.operationType_ !=
+          common::DataBaseRequest::OperationType::Insert &&
+      isPrimaryKeyEmpty(requestData.primaryKey_))
+    return {};
+
+  if (requestData.operationType_ !=
+          common::DataBaseRequest::OperationType::Select &&
+      isDataEmpty(requestData.data_))
```

Comment:
you're returning an empty string if the primary key is empty or if the data is empty. This might be a valid approach, but it would be better to throw an exception or return an error code to indicate that something went wrong.

Reply:
You right, Maybe we can move it to another task. For example FaultHandling DataBase. We can talk about those fault scenarios on meeting? But for now I think we should delivere this version to unblock another tasks.
What do you think?